diffusion/models/stable_diffusion.py

```diff
@@ -142,52 +143,52 @@ def set_rng_generator(self, rng_generator: torch.Generator):
         self.rng_generator = rng_generator
 
     def forward(self, batch):
-        latents, conditioning, conditioning_2, pooled_conditioning = None, None, None, None
+        latents, text_embeds, text_pooled_embeds, attention_mask, encoder_attention_mask = None, None, None, None, None
+        if 'attention_mask' in batch:
+            attention_mask = batch['attention_mask']  # mask for text encoders
+            # text mask for U-Net
+            if self.mask_pad_tokens:
+                if len(attention_mask.shape) == 2:
+                    encoder_attention_mask = attention_mask
+                elif len(attention_mask.shape) == 3:
+                    encoder_attention_mask = attention_mask[:, 0]
+                    for i in range(1, attention_mask.shape[1]):
```

Comment:
This should probably be done in dataloading. No need to change it for this PR though.

Reply:
Yeah, that makes sense 😓